File: roman-to-integer/roman-to-integer.py
class Solution:
    def roman_to_integer(self, s: str) -> int:
        symbols = [
            ('IV', 4),
            ('IX', 9),
            ('XL', 40),
            ('XC', 90),
            ('CD', 400),
            ('CM', 900),
            ('I', 1),
            ('V', 5),
            ('X', 10),
            ('L', 50),
            ('C', 100),
            ('D', 500),
            ('M', 1000)
        ]
        result = 0
        for j, k in symbols:
            if j in s:
                result += s.count(j) * k
                s = s.replace(j, '')
        return result

test = Solution()
print("III = " + str(test.roman_to_integer("III")))
print("IV = " + str(test.roman_to_integer("IV")))
print("IX = " + str(test.roman_to_integer("IX")))
print("LVIII = " + str(test.roman_to_integer("LVIII")))
print("MCMXCIV = " + str(test.roman_to_integer("MCMXCIV")))

File: roman-to-integer/RomanToInteger.java
import java.util.HashMap;

public class RomanToInteger {

	public int RomanToIntegerSolution(String s) {
		int result = 0;
		char[] charArray = s.toCharArray();
		
		HashMap<Character,Integer> symbols = new HashMap<>();
		symbols.put('I', 1);
		symbols.put('V', 5);
		symbols.put('X', 10);
		symbols.put('L', 50);
		symbols.put('C', 100);
		symbols.put('D', 500);
		symbols.put('M', 1000);
		
		for (int i = 0; i < charArray.length; i++) {
			if(i == charArray.length - 1) {
				result += symbols.get(charArray[i]);
			} else if (symbols.get(charArray[i]) >= symbols.get(charArray[i + 1])) {
				result += symbols.get(charArray[i]);
			} else {
				result += symbols.get(charArray[i + 1]) - symbols.get(charArray[i]);
				i++;
			}
		}
		return result; 
	    }
}
